src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -204,11 +204,10 @@
   timeout. If the timeout runs out then nil will be returned, indicating that
   there were no instances available."
   [pool-context :- jruby-schemas/PoolContext
-   timeout :- schema/Int
```

Comment:
So... I agree that it's weird that there was a borrow timeout in the context and it was also passed in here, but, this change seems like it forces consumers to only be able to have a single borrow timeout that can ever be used.

I can imagine a case where a consumer might have different types of uses for the JRuby instances, with different priorities / urgencies / failure conditions.  So they might want to have a really long borrow timeout for some of those cases and a really short one for others, and this change precludes that.

Is the copy of the borrow timeout that is in the context used elsewhere?  If so, where?  Maybe it would make sense to remove it from the config instead of removing it from this signature?

Was there anything else that motivated this change, or was it just an inconsistency that you noticed?

I don't want to drag this work out so I'm probably ok with this as-is; we could always add in another arity for this fn in the future if the use case that I described above were to come to pass.  But I am at least interested in knowing why the timeout is in the config.